WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -114,48 +133,56 @@ private extension ItemListView {
-                                    try? await Task.sleep(nanoseconds: 300 * NSEC_PER_MSEC)
-                                }
-
-                                guard !Task.isCancelled else { return }
-
-                                didFinishSearch = false
-
-                                await posModel.purchasableItemsSearchController.searchItems(searchTerm: newValue, baseItem: .root)
-
-                                if !Task.isCancelled {
-                                    didFinishSearch = true
+                    if isSearchAllowed {
+                        searchField
+                            .renderedIf(shouldShowSearchField)
+                            .transition(.opacity.combined(with: .move(edge: .trailing)))
+                            .onChange(of: searchTerm) { oldValue, newValue in
+                                // The debouncing logic is a little tricky, because the loading state is held in the controller.
+                                // Arguably, we should use view state `isSearching` for this, so the UI is independent of the request timing.
+
+                                // As the user types, we don't want to send every keystroke to the remote, so we debounce the requests.
+                                // However, we don't want to debounce the first keystroke of a new search, so that the loading
+                                // state shows immediately and the UI feels responsive.
+
+                                // So, if the last search was finished, we don't debounce the first character. If it didn't
+                                // finish i.e. it is still ongoing, we debounce the next keystrokes by 300ms. In either case,
+                                // the ongoing search is redundant now there's a new search term, so we cancel it.
+                                let shouldDebounceNextSearchRequest = !didFinishSearch
+                                searchTask?.cancel()
+
+                                searchTask = Task {
+                                    if shouldDebounceNextSearchRequest {
+                                        try? await Task.sleep(nanoseconds: 300 * NSEC_PER_MSEC)
+                                    }
+
+                                    guard !Task.isCancelled else { return }
+
+                                    didFinishSearch = false
+
+                                    await posModel.purchasableItemsSearchController.searchItems(searchTerm: newValue, baseItem: .root)
+
+                                    if !Task.isCancelled {
+                                        didFinishSearch = true
+                                    }
                                 }
                             }
+
+                        POSPageHeaderActionButton(systemName: "magnifyingglass") {
+                            selectedItemListType = .products(search: true)
+                            isSearchFieldFocused = true
```

Comment:
The keyboard doesn't appear automatically the first time I tap the search button and there are no recent searches.

It appears automatically after I add the first recent search.

iPad Air M2 iOS 18.3

https://github.com/user-attachments/assets/5700dd67-6812-46ba-95c3-6968dc59891a

Reply:
That's very strange... and I can't reproduce it.

Here's 18.4 on a simulator (you have to have used the software keyboard for the previous text entry in that case):

https://github.com/user-attachments/assets/fb7e9f4d-f120-4c09-826a-3489e38d2087

It's the same on iOS 18.2 simulator, and on 17.7.3 on my test iPad. I can't find an 18.3 simulator.

Is this 100%, or difficult to reproduce?

Reply:
It's quite hard to understand. Maybe it's some OS or SwiftUI bug? It's not 100% reproducible. At first, the keyboard opened, then I dismissed it, and after that, it wouldn't appear automatically anymore. I went to 'Menu -> Get Support' which triggered keyboard opening. After that, the keyboard continued to open even if I dismissed it.

https://github.com/user-attachments/assets/743507e5-353d-4659-8f87-ea377c36467c

I do have a physical Bluetooth keyboard for my iPad, but I tested it now with Bluetooth turned off. 

I think we can create a lower-priority ticket for later. I'll investigate it myself if you don't manage to reproduce it yourself.

Reply:
@staskus Yeah, the hardware keyboard was my thought as well. Another potential culprit is Continuity, if you have it signed in to the same Apple Account as your Mac.

For me, the keyboard is shown immediately and consistently... but I'll hook up a keyboard and try a little more to recreate, and make an issue if I can't. 

We already specifically set the text field as focused when the button's tapped. There may be a timing issue with that on iOS 18, but it's a `@State` of the container view, so it should be respected regardless of the timing, really.